### DISCUSSION TITLE
Tile creation failure with "SSL library has no ciphers"

### DISCUSSION THREAD

**Discussion by philvarner** [cogeotiff/rio-tiler#582] [Q&A] [answered]
I'm using an old fork of titiler 0.3.5 (https://github.com/NASA-IMPACT/titiler) running in a Lambda w/ Python 3.8, I'm seeing a frequent issue where the generation of a tile fails with the following log messages. I know this is somehow related to the use of OpenSSL 1.0.2k in the Lambda environment and it not getting initialized properly, but I've been unable to find a solution. I was hoping this had been seen before.

The error message is:

```
CURL error: SSL: couldn't create a context: error:140A90A1:SSL routines:func(169):reason(161)
NoneType: None
CURL error: SSL: couldn't create a context: error:140A90A1:SSL routines:SSL_CTX_new:library has no ciphers
NoneType: None
CURL error: SSL: couldn't create a context: error:140A90A1:SSL routines:SSL_CTX_new:reason(161)
NoneType: None
```

This happens for maybe 1 out of 10 tile requests, and it's random as executing the same request again works.

**Comment by vincentsarago** [comment, marked as answer]
this must be a gdal or rasterio issue not a rio-tiler one. I'll convert this thread to discussion

**Comment by philvarner** [comment]
I think you're right about it rasterio or gdal, so moved here https://github.com/rasterio/rasterio/discussions/2784
